File: src/main/java/com/example/glamourstudio/entities/Servico.java
package com.example.glamourstudio.entities;


import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Servico{

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String nome;

    private String descricao;

    private BigDecimal valor;

    private Integer duracaoMinutos;

    private EnumStatusServico statusServico = EnumStatusServico.ATIVO;

}

File: front/app/types/servico.ts
export class Servico{
    constructor(
        public id:number | null,
        public nome:string,
        public descricao:string,
        public valor:number,
        public duracaoMinutos:number,
        public status:string
    ){}
}

export interface ServicoFormProps{
    servicoExistente?:Servico
}